internal/agentlint/agentlint_test.go

```
@@ -0,0 +1,382 @@
+
+// collectFiles returns all files under the given directories that match the
+// given extension filter.  It returns paths relative to root for display, and
+// absolute paths for reading.
+func collectFiles(t *testing.T, root string, dirs []string, ext string, skipDirs []string) []string {
+	t.Helper()
+	var files []string
+	for _, dir := range dirs {
+		absDir := filepath.Join(root, dir)
+		err := filepath.Walk(absDir, func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return nil // skip unreadable
```

golangci-lint (2.11.4)

[error] error is not nil (line 219) but it returns nil

(nilerr)